File: src/CookBook/Models/Ingredient.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CookBook.Models
{
    public class Ingredient : IEntityBase
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public int DisplayOrder { get; set; }
        public DateTime CreatedOn { get; set; }
        public virtual Recipe Recipe { get; set; }
        public int RecipeId { get; set; }
    }
}

File: src/CookBook/wwwroot/app/models/ingredient.ts
export class Ingredient {

    Id: number;
    Title: string;
    Description: string;
    CreatedOn: Date;

    constructor(id: number, title: string, description: string, createdOn: Date)
    {
        this.Id = id;
        this.Title = title;
        this.Description = description;
        this.CreatedOn = createdOn;
    }
}